### ISSUE TITLE
[Accessibility] Semantics `header` property is ignored on Android

### ISSUE THREAD

**Issue by JJdeGroot** [flutter/flutter#179678]
### Steps to reproduce

1. Create a text view, marked as header:
```dart
Semantics(
  header: true,
  child: Text('Descriptive header')
);
```
2. Install on Android device
3. Navigate to the text view using TalkBack

This commit: https://github.com/flutter/flutter/commit/59d8010db5139da19b265c3f7cf29f90a86b8e5e has changed the logic for setting the `isHeading` property.

`result.setHeading(semanticsNode.hasFlag(Flag.IS_HEADER));` was replaced by `result.setHeading(semanticsNode.headingLevel > 0);` without a fallback.

Expected behavior would be to set `isHeading` to true if `Flag.IS_HEADER` OR `headingLevel > 0`.

### Expected results

TalkBack should announce the text view as header.

### Actual results

TalkBack does not announce is header.

### Code sample

```dart
Semantics(
  header: true,
  child: Text('Descriptive header')
);
```

### Screenshots or Video

The `isHeading` property is false in the accessibility tree.

<img width="1867" height="1057" alt="Accessibility tree for Android app built in Flutter" src="https://github.com/user-attachments/assets/00c4c0c5-afeb-475d-9a58-16ead4e4f0ad" />

### Logs

N/A

### Flutter Doctor output

N/A

**Comment by darshankawar** [comment]
Thanks for the report and potential fix.

**Comment by JJdeGroot** [comment]
@chunhtai @darshankawar I strongly disagree with your interpretation.

**How is there no equivalent to Flutter's header in Android?**

Flutter docs for [SemanticProperties#header](https://api.flutter.dev/flutter/semantics/SemanticsProperties/header.html):

> If non-null, indicates that this subtree represents a header.
> A header divides into sections. For example, an address book application might define headers A, B, C, etc. to divide the list of alphabetically sorted contacts into sections.

Android docs for [AccessibilityNodeInfo#isHeading()](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#isHeading()):

> Returns whether node represents a heading.
> Note: Returns true if either [setHeading(boolean)](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#setHeading(boolean)) marks this node as a heading or if the node has a [CollectionItemInfo](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo.CollectionItemInfo) that marks it as such, to accomodate apps that use the now-deprecated API.

iOS docs for [UIAccessibilityTraits#header](https://developer.apple.com/documentation/uikit/uiaccessibilitytraits/header):

> The accessibility element is a header that divides content into sections, such as the title of a navigation bar.

My interpretation is that Flutter's semantic header maps 1:1 to Android's accessibility isHeading and iOS its accessibility header properties.

The iOS codebase also still maps `flags.isHeader` to the `UIAccessibilityTraitsHeader` property 

https://github.com/flutter/flutter/blob/9a329d665d87153be430461a36ff0604b2ae6c2f/engine/src/flutter/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L842

Do you also consider this to be a mistake?

**Comment by chunhtai** [comment]
> Do you also consider this to be a mistake?

No, iOS has concept of header in their API, so that part is correct.

Flutter originally mixed up the concept of `header` and `heading`, we originally thought they are the same thing, until we expand to different platforms and figured out they are two separate concepts.

**Comment by zywek123** [comment]
For example, I haven’t seen anywhere in any Android interface—with the exception of web interfaces—where TalkBack could read a heading as level 2, level 1, or level x. Setting `header: true` simply added semantic information indicating that it was a heading. And setting heading levels has always been, and will long remain, associated with web interfaces.

**Comment by chunhtai** [comment]
Sorry my previous response was vague

## Current behavior

### Semantics(header: true)

in web, this becomes aria-role="banner".  (which is the current role for header according to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/banner_role)

in iOS, this translate to [AccessibilityTraitHeader](https://developer.apple.com/documentation/uikit/uiaccessibilitytraits/header?language=objc)

in Android, noop

in desktop, this translate to banner

### Semantics(headingLevel: 1)

in web, this becomes h1

in iOS, not wired up yet, but it should probably be https://developer.apple.com/documentation/foundation/nsattributedstring/key/accessibilitytextheadinglevel

in Android, this sets setHeading

in desktop, not wired up yet

**Comment by chunhtai** [comment]
> I haven’t seen anywhere in any Android interface—with the exception of web interfaces—where TalkBack could read a heading as level 2, level 1, or level x. Setting header: true simply added semantic information indicating that it was a heading

even though number don't make sense in android, we still shouldn't co-relate header to be heading in android. I think if we really want a deterministic API, we should use Semantics(heading:true), but that feels a bit unecessary if all developer needs to do is to give a non 0 value to the existing semantics heading API

**Comment by chunhtai** [comment]
will put up a breaking change doc for this, I think the original pr missed this when it first merge

**Comment by JJdeGroot** [comment]
I can agree with the reasons to align heading and header semantics.

If header is meant to indicate banner role, then iOS mapping should also be noop after wiring up heading levels?

Also when wiring that up, don't forget that both header trait and heading level need to be set to have VoiceOver read out heading levels.

I agree that Android heading:true semantics is not needed considering headingLevel api is available and does set isHeading=true for now and potentially heading levels if Android supports that in the future. 

Overall, what's most needed is the breaking change doc, as this was a breaking change on Android. 

Also on iOS, if header semantics will no longer set header trait after wiring up heading levels api, this also needs breaking change doc. 

It would be nice to raise prio of wiring up heading levels API, to have consistent behavior of the heading level semantics on Android and iOS.

**Comment by chunhtai** [comment]
I will work on wiring up heading level. I am still not sure about header trait though. I have to look more into it

**Comment by chunhtai** [comment]
You are right, looks like ios treats header trait as heading, which is sad. I will also update the ios embedding to owner headlevel instead of header
